libs/brain/command/src/lib/brn-command.directive.ts

```diff
@@ -0,0 +1,77 @@
+import { ActiveDescendantKeyManager } from '@angular/cdk/a11y';
+import {
+	AfterViewInit,
+	computed,
+	contentChild,
+	contentChildren,
+	Directive,
+	HostListener,
+	inject,
+	Injector,
+	input,
+} from '@angular/core';
+import { BrnCommandItemToken } from './brn-command-item.token';
+import { BrnCommandSearchInputDirective } from './brn-command-search-input.directive';
+import { provideBrnCommand } from './brn-command.token';
+
+@Directive({
+	selector: '[brnCommand]',
+	standalone: true,
+	providers: [provideBrnCommand(BrnCommandDirective)],
+	host: {
+		'[id]': 'id()',
+	},
+})
+export class BrnCommandDirective implements AfterViewInit {
+	private static _id = 0;
+
+	private readonly _injector = inject(Injector);
+
+	/** The id of the command */
+	public readonly id = input<string>(`brn-command-${BrnCommandDirective._id++}`);
+
+	/** The default filter function */
+	private readonly _defaultFilter = (value: string, search: string) =>
+		value.toLowerCase().includes(search.toLowerCase());
+
+	/** A custom filter function to use when searching. */
+	public readonly filter = input<CommandFilter>(this._defaultFilter);
+
+	/** @internal The search query */
+	public readonly search = computed(() => this._searchInput()?.value() ?? '');
+
+	/** Access the search input if present */
+	private readonly _searchInput = contentChild(BrnCommandSearchInputDirective, {
+		descendants: true,
+	});
+
+	/** @internal Access all the items within the commmand */
+	public readonly items = contentChildren(BrnCommandItemToken, {
+		descendants: true,
+	});
+
+	/** @internal The key manager for managing active descendant */
+	public readonly keyManager = new ActiveDescendantKeyManager(this.items, this._injector);
+
+	constructor() {
+		this.keyManager
+			.withVerticalOrientation()
+			.withHomeAndEnd()
+			.withWrap()
+			.withTypeAhead()
+			.skipPredicate((item) => item.disabled || !item.visible());
+	}
+
+	ngAfterViewInit(): void {
+		if (this.items().length) {
+			this.keyManager.setActiveItem(0);
+		}
+	}
```

Comment:
This causes an error in SSR. Could this be replaced with `afterNextRender` only applying in the browser?